#### Task (class ‘B’):

    Compare the global indices extracted from PDC and DTF connectivity estimations.

#### Importing the routine libraries

In [5]:
import sys
sys.path.insert(0, '../Lib')
import homeworkLib as hwl
import pickle
import networkx as nx

### Loading our graphs from question 1.2

In [6]:
with open("../Pickle/EO_12.file", "rb") as f:
    EO = pickle.load(f)
with open("../Pickle/EC_12.file", "rb") as f:
    EC = pickle.load(f)

# EO: Eyes-opened case
dtf_A_EO = EO.bin_adj_matrix_dtf
pdc_A_EO = EO.bin_adj_matrix_pdc

dtf_graph_EO = EO.dtf_graph
pdc_graph_EO = EO.pdc_graph

dtf_matrix_EO = EO.dtf_matrix
pdc_matrix_EO = EO.pdc_matrix

# EC: Eyes-closed case
dtf_A_EC = EC.bin_adj_matrix_dtf
pdc_A_EC = EC.bin_adj_matrix_pdc

dtf_graph_EC = EC.dtf_graph
pdc_graph_EC = EC.pdc_graph

dtf_matrix_EC = EC.dtf_matrix
pdc_matrix_EC = EC.pdc_matrix

### Global graph indices

#### Eyes opened case

In [7]:
print("Clustering coefficient of the dtf graph: ", nx.average_clustering(dtf_graph_EO))
print("Clustering coefficient of the pdc graph: ", nx.average_clustering(pdc_graph_EO))

print("Average path length of the dtf graph: ", nx.average_shortest_path_length(dtf_graph_EO))
print("Average path length of the pdc graph: ", nx.average_shortest_path_length(pdc_graph_EO))

Clustering coefficient of the dtf graph:  0.5348678449063254
Clustering coefficient of the pdc graph:  0.3109809978504136
Average path length of the dtf graph:  0.8732638888888888
Average path length of the pdc graph:  1.7886904761904763


#### Eyes closed case

In [8]:
print("Clustering coefficient of the dtf graph: ", nx.average_clustering(dtf_graph_EC))
print("Clustering coefficient of the pdc graph: ", nx.average_clustering(pdc_graph_EC))

print("Average path length of the dtf graph: ", nx.average_shortest_path_length(dtf_graph_EC))
print("Average path length of the pdc graph: ", nx.average_shortest_path_length(pdc_graph_EC))

Clustering coefficient of the dtf graph:  0.47227191755347286
Clustering coefficient of the pdc graph:  0.33039512558322603
Average path length of the dtf graph:  0.7626488095238095
Average path length of the pdc graph:  2.1646825396825395


### Conclusions and comparisons

#### Eyes-opened case: PDC vs DTF

* Clustering coefficient

Let's start by talking about the clustering coefficient (CC). We observed that the CC of the DTF network (.53) was considerably greater than that of PDC network (.31). the clustering coefficient of a node is a measure of the probability that two randomly selected "friends" of that node are "friends" themselves. Ergo, a high average clustering coefficient for a network suggest the ability of that network to highlight group-forming. It then turns out that the DTF MVAR estimator better highlights the connections among brain areas performing related functions than the PDC MVAR estimator.

* Average path length

Contrary to what has been observed in the CC case, the average path length of the PDC network (1.79) was slightly higher than twice that of the DTF network (.87). The average path length represents the average of the shortest path length, averaged over all pairs of nodes. a low average path length for a network suggests that the network is able to highlight all the shortest possible connections among nodes. In our case, the DTF network was better having mean path length of half that of the PDC network.

#### Eyes-closed case: PDC vs DTF

In this case, for both the clustering coefficient and the average path length, the situation was quite similar to the eyes opened case, namely the the DTF MVAR estimator performed better than the PDC estimator.